server/plugin/api_test.go

```diff
@@ -249,74 +265,232 @@ func TestPlugin_handleVirtualAgentWebhook(t *testing.T) {
-							Text:  "mockText",
-							Typed: true,
-						},
+				URL:    fmt.Sprintf("/api/v1%s?secret=mockWebhookSecret", PathVirtualAgentWebhook),
+				Body: `{
+					"requestId": "9ff925c1-893f-4a46-baa5-a64f4d8c89b1",
+					"message": {
+					  "text": "",
+					  "typed": true
 					},
-					Body: []MessageResponseBody{
+					"userId": "6816f79cc0a8016401c5a33be04be441",
+					"body": [
+					  {
+						"uiType": "OutputLink",
+						"group": "DefaultText",
+						"label": "Successful",
+						"header": "header",
+						"value": {
+							"action": "action"
+						}
+					  }
+					],
+					"score": 1
+				  }`,
+			},
+			expectedResponse: testutils.ExpectedResponse{
+				StatusCode: http.StatusOK,
+			},
+		},
+		"TopicPickerControl response is received from Virtual Agent": {
+			httpTest: httpTestString,
+			request: testutils.Request{
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("/api/v1%s?secret=mockWebhookSecret", PathVirtualAgentWebhook),
+				Body: `{
+					"requestId": "9ff925c1-893f-4a46-baa5-a64f4d8c89b1",
+					"message": {
+					  "text": "",
+					  "typed": true
+					},
+					"userId": "6816f79cc0a8016401c5a33be04be441",
+					"body": [
+					  { 
+						"uiType":"TopicPickerControl", 
+						"group":"DefaultPicker", 
+						"nluTextEnabled":false, 
+						"promptMsg":"Hi guest, please enter your request or make a selection of what I can help with. You can type help any time when you need help.", 
+						"label":"Show me everything", 
+						"options":[ 
+						  { 
+							"label":"b2b topic", 
+							"value":"2bb7bd7670de6010f877c7f188266fc7", 
+							"enabled":true 
+						  }, 
+						  { 
+							 "label":"Live Agent Support.", 
+							 "value":"ce2ee85053130010cf8cddeeff7b12bf", 
+							 "enabled":true 
+						  } 
+						] 
+					  }
+					],
+					"score": 1
+				  }`,
+			},
+			expectedResponse: testutils.ExpectedResponse{
+				StatusCode: http.StatusOK,
+			},
+		},
+		"OutputText response is received from Virtual Agent": {
+			httpTest: httpTestString,
+			request: testutils.Request{
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("/api/v1%s?secret=mockWebhookSecret", PathVirtualAgentWebhook),
+				Body: `{
+					"requestId": "9ff925c1-893f-4a46-baa5-a64f4d8c89b1",
+					"message": {
+					  "text": "",
+					  "typed": true
+					},
+					"userId": "6816f79cc0a8016401c5a33be04be441",
+					"body": [
+					  {
+						"uiType": "OutputText",
+						"group": "DefaultText",
+						"value": "Successful",
+						"maskType": "NONE"
+					  }
+					],
+					"score": 1
+				  }`,
+			},
+			expectedResponse: testutils.ExpectedResponse{
+				StatusCode: http.StatusOK,
+			},
+		},
+		"Picker response is received from Virtual Agent": {
+			httpTest: httpTestString,
+			request: testutils.Request{
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("/api/v1%s?secret=mockWebhookSecret", PathVirtualAgentWebhook),
+				Body: `{
+					"requestId": "9ff925c1-893f-4a46-baa5-a64f4d8c89b1",
```

Comment:
Here and in every test case id in the body if you aren't matching should not be close to the actual value.
```suggestion
					"requestId": "test-requestId",
```